pkg/common/ingress/controller/backend_ssl_test.go

```diff
@@ -143,92 +124,70 @@ func buildCrtKeyAndCA() ([]byte, []byte, []byte, error) {
 func TestSyncSecret(t *testing.T) {
 	// prepare for test
 	dCrt, dKey, dCa, err := buildCrtKeyAndCA()
-	if err != nil {
-		t.Fatalf("Unexpected error: %v", err)
-	}
+	assert.Nil(t, err, "unexpected error: %v", err)
 
-	foos := []struct {
-		tn            string
+	cases := map[string]struct {
 		secretName    string
 		Data          map[string][]byte
 		expectSuccess bool
 	}{
-		{"getPemCertificate_error", "default/foo_secret", map[string][]byte{apiv1.TLSPrivateKeyKey: dKey}, false},
-		{"normal_test", "default/foo_secret", map[string][]byte{apiv1.TLSCertKey: dCrt, apiv1.TLSPrivateKeyKey: dKey, tlscaName: dCa}, true},
+		"getPemCertificate_error": {"default/foo_secret", map[string][]byte{apiv1.TLSPrivateKeyKey: dKey}, false},
+		"normal_test":             {"default/foo_secret", map[string][]byte{apiv1.TLSCertKey: dCrt, apiv1.TLSPrivateKeyKey: dKey, tlsCaName: dCa}, true},
 	}
 
-	for _, foo := range foos {
-		t.Run(foo.tn, func(t *testing.T) {
-			ic := buildGenericControllerForBackendSSL()
-
+	for n, c := range cases {
+		t.Run(n, func(t *testing.T) {
 			// init secret for getPemCertificate
-			secret := buildSecretForBackendSSL()
-			secret.SetNamespace("default")
-			secret.SetName("foo_secret")
-			secret.Data = foo.Data
-			ic.listers.Secret.Add(secret)
-
-			key := "default/foo_secret"
-			// for add
-			ic.syncSecret(key)
-			if foo.expectSuccess {
-				// validate
-				_, exist := ic.sslCertTracker.Get(key)
-				if !exist {
-					t.Errorf("Failed to sync secret: %s", foo.secretName)
-				} else {
-					// for update
-					ic.syncSecret(key)
-				}
-			}
+			secret := buildSecretForBackendSSL("default", "foo_secret")
+			secret.Data = c.Data
+
+			cs := fake.NewSimpleClientset()
+			l := newListers(newIngressListenerForBackendSSL(cs), newSecretListerForBackendSSL(cs, secret))
+			ic := buildGenericControllerForBackendSSL(cs, l)
+			ic.syncSecret(c.secretName)
+
+			_, e := Get(c.secretName)
```

Comment:
Travis and my local build complained with this line. Changing to this fixes the error:

```golang
			_, e := ic.sslCertTracker.Get(c.secretName)
```